modules/core/src/main/java/org/apache/ignite/internal/processors/cache/ClusterCachesInfo.java

```diff
@@ -1084,6 +1085,15 @@ else if (encMgr.masterKeyDigest() != null &&
             }
         }
 
+        if (containsInvalidFileNameChars(ccfg, ctx.config().getDataStorageConfiguration())) {
```

Comment:
I see that this change was originally proposed by Nikita under [IGNITE-20867](https://issues.apache.org/jira/browse/IGNITE-20867), however, it is a bit weird to me that we postponed this validation on this phase, I mean this validation will be done in the discovery thread when a DynamicCacheChangeRequest is received by server node. This can lead to hanged PME in the following case:
 - let's assume we have 2 nodes cluster: the first node is the coordinator on Linux and the second one is the server node on Windows.
  - start cache with the name that contains an invalid symbol on Windows only.
  - the coordinator will successfully process the message and trigger PME
  - the second node will reject the request.
  
Let's try to check this case. 
Perhaps, the validation can be moved to another place, please take a look at IgniteKernal.getOrCreateCache0 -> CU.validateNewCacheName(cacheName);

Reply:
I moved it to GridCacheProcessor#dynamicStartCache

Reply:
After a private discussion it was decided to move this validation to `CU.validateNewCacheName`